Sources/SwiftUICoordinator/DeepLink/DeepLink.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  DeepLink.swift
+//  SwiftUICoordinatorExample
+//
+//  Created by Erik Drobne on 11/09/2023.
+//
+
+import Foundation
+
+public struct DeepLink {
+    public let action: String
+    public let route: NavigationRoute
+    public let params: Set<String>
+    
+    public init(action: String, route: NavigationRoute, params: Set<String> = []) {
+        self.action = action
+        self.route = route
+        self.params = params
+    }
+}
+
+extension DeepLink: Hashable {
+    public static func == (lhs: DeepLink, rhs: DeepLink) -> Bool {
+        lhs.action == rhs.action
```

Comment:
not sure if it's possible, but what if there are two deep links with same action, but different parameters?

Reply:
I would treat two links with the same action and different parameters as the same.
They still have the same route to navigate, except the parameters are different, which
should be handled in the `handle` method later on.

Example:

`myapp://paywall?offer=monthly`
`myapp://paywall?offer=annual`

Both links should lead to the paywall screen, where the appearance of the prominent offer is
either monthly or annual.